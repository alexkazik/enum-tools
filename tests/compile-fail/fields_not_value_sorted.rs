#![no_implicit_prelude]

use ::enum_tools::EnumTools;

#[derive(EnumTools, Debug, Clone, Copy, Eq, PartialEq)]
#[enum_tools(sorted(value))]
#[repr(i8)]
pub(crate) enum EG {
    #[enum_tools(rename = "A*")]
    A = 0,
    B = 1,
    C = 2,
    D = -3,
}

fn main() {}
