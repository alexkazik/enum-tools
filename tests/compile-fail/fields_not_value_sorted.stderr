error: variants are not sorted by value
  --> tests/compile-fail/fields_not_value_sorted.rs:13:5
   |
13 |     D = -3,
   |     ^
